setup.py

```diff
@@ -1,7 +1,67 @@
 # Copyright Pincer 2021-Present
 # Full MIT License can be found in `LICENSE` at the project root.
 
-from setuptools import setup
+from setuptools import setup, find_packages
+import re
 
-if __name__ == '__main__':
-    setup()
+with open("src/pincer/__init__.py") as f:
+    version = re.search(
+        r'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]', f.read(), re.MULTILINE
+    ).group(1)
+
+requirements = []
+with open("requirements.txt") as f:
+    requirements = f.read().splitlines()
+
+extra_requires = {
+    "img": [
+        "Pillow>=8.4.0",
+    ],
+    "speed": [
+        "orjson>=3.5.4",
+        "Brotli>=1.0.9",
+        "aiodns>=1.1",
+        "cchardet>=2.1.7",
+    ],
+}
+
+setup(
+    name="pincer",
+    version=version,
+    packages=find_packages(where="src"),
+    package_dir={"": "src"},
+    project_urls={
+        "Website": "https://pincer.dev",
+        "Documentation": "https://docs.pincer.dev",
+        "Discord": "https://discord.gg/8WkYz3fNFm",
+        "Issue Tracker": "https://github.com/pincer-org/pincer/issues",
+        "Pull Request Tracker": "https://github.com/pincer-org/piner/pulls",
+    },
+    url="https://github.com/pincer-org/pincer",
+    license="MIT",
+        author="Sigmanificient, Arthurdw",
+    author_email="contact@pincer.dev",
+    long_description=open("README.md").read(),
+    long_description_content_type="text/markdown",
+    install_requires=requirements,
+    extra_requires=extra_requires,
+    description="Discord API wrapper rebuild from scratch.",
+    python_requires=">=3.8",
+    classifiers=[ # https://pypi.org/classifiers
+        "Development Status :: 2 - Pre-Alpha",
+        "License :: OSI Approved :: MIT License",
+        "Intended Audience :: Developers",
+        "Natural Language :: English",
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Framework :: AsyncIO",
+        "Framework :: aiohttp",
+        "Topic :: Internet",
+        "Topic :: Software Development :: Libraries",
+        "Topic :: Software Development :: Libraries :: Python Modules",
+        "Topic :: Utilities",
+    ],
+)
```

Comment:
I wouldn't change the `setup.cfg` to a `setup.py` file as it run literally run anything during the install process, and put a load of uneeded logic within the `setup.py` ifself.

>  ​``setup.cfg``  is a cheekily named Python package which supports providing all of a Python distribution's metadata and build configuration via the ​``setup.cfg``​ ​* file  at the base of the distribution's source tree, rather ​than in the ​``setup.py``​ script. 
    The standard ``setup.py`` script is reduced to a stub which uses the ``setup.cfg`` package to convert the contents of the ``setup.cfg``​ file to keyword arguments understood by the ``setup()``​ function (in particular, the implementation provided by setuptools).  Thus all the standard functionality of ​the ​``setup.py`` script is still supported for tools and procedures that use ​it. 
   ​  In other words, rather than defining your project in a source file, it is defined in an easy to parse, easy to read plain-text file.  However, several hook points are still provided where Python functions can modify the build process if necessary.  But efforts are made to minimize the need for this in the majority of Python projects.

[More](https://docs.python.org/3/distutils/configfile.html)

Reply:
using a setup.cfg and the many differences are just pretty hard imo, this is both easier to read and modify, and most of the pluses of using a setup.cfg instead doesn't seem like anything we would need to care about?